new_platforms/opteesim/Trusted/cyres_sim.c

```diff
@@ -0,0 +1,455 @@
+        dice_measurement,
+        DICE_DIGEST_LENGTH);
+
+    /* Don't use identity directly */
+    RiotCrypt_Hash(digest, RIOT_DIGEST_LENGTH, id->fwid, DICE_DIGEST_LENGTH);
+
+    /* Derive ID key pair from identity hash */
+    RIOT_STATUS riotStatus = RiotCrypt_DeriveEccKey(
+        &id->pub,
+        &id->priv,
+        digest,
+        RIOT_DIGEST_LENGTH,
+        RIOT_LABEL_IDENTITY,
+        lblSize(RIOT_LABEL_IDENTITY));
+    oe_assert(riotStatus == RIOT_SUCCESS);
+}
+
+/* Construct and sign a device certificate */
+static void build_device_cert(
+    DERBuilderContext* der_ctx,
+    simulated_cyres_identity_t* id,
+    simulated_cyres_identity_t* signer,
+    RIOT_ECC_PUBLIC* code_authority,
+    RIOT_X509_TBS_DATA* tbs)
+{
+    /* Build the device identity */
+    uint8_t device_id[CLAIM_BUFFER_LENGTH];
+    uint32_t device_id_len = 0;
+    uint8_t auth_buffer[65];
+    uint32_t auth_size;
+
+    RiotCrypt_ExportEccPub(code_authority, auth_buffer, &auth_size);
+    RIOT_STATUS status = BuildDeviceClaim(
+        &id->pub,
+        auth_buffer,
+        auth_size,
+        NULL,
+        0,
+        device_id,
+        CLAIM_BUFFER_LENGTH,
+        &device_id_len);
+    oe_assert(status == RIOT_SUCCESS);
+
+    /* Create the device certificate */
+    X509GetDeviceCertTBS(
+        der_ctx, tbs, &id->pub, NULL, device_id, device_id_len, 1);
+
+    /* Sign the Certificate's TBS region and create the final DER cert */
+    RIOT_ECC_SIGNATURE tbs_sig = {0};
+    RiotCrypt_Sign(&tbs_sig, der_ctx->Buffer, der_ctx->Position, &signer->priv);
+    X509MakeAliasCert(der_ctx, &tbs_sig);
+}
+
+/* Construct and sign an alias certificate */
+static void build_alias_cert(
+    DERBuilderContext* der_ctx,
+    simulated_cyres_identity_t* id,
+    simulated_cyres_identity_t* signer,
+    RIOT_ECC_PUBLIC* code_authority,
+    RIOT_X509_TBS_DATA* tbs)
+{
+    /* Build the claim blob */
+    uint8_t alias_claim[CLAIM_BUFFER_LENGTH];
+    uint32_t alias_claim_len = 0;
+    uint8_t auth_buffer[65];
+    uint32_t auth_size;
+
+    RiotCrypt_ExportEccPub(code_authority, auth_buffer, &auth_size);
+    RIOT_STATUS status = BuildAliasClaim(
+        auth_buffer,
+        auth_size,
+        id->fwid,
+        sizeof(id->fwid),
+        alias_claim,
+        CLAIM_BUFFER_LENGTH,
+        &alias_claim_len);
+    oe_assert(status == RIOT_SUCCESS);
+
+    /* Create the alias certificate */
+    X509GetAliasCertTBS(
+        der_ctx,
+        tbs,
+        &id->pub,
+        &signer->pub,
+        id->fwid,
+        sizeof(id->fwid),
+        alias_claim,
+        alias_claim_len,
+        0);
+
+    /* Sign the Certificate's TBS region */
+    RIOT_ECC_SIGNATURE tbs_sig = {0};
+    RiotCrypt_Sign(&tbs_sig, der_ctx->Buffer, der_ctx->Position, &signer->priv);
+    X509MakeAliasCert(der_ctx, &tbs_sig);
+}
+
+/* Generate a PEM encoded certificate of the type specified for an identity.
+    The certificate will be issued by the provided signer identity. */
```

Comment:
nit: comment style is odd.  Compare lines 291-293

Reply:
fixed